justfile

```diff
@@ -130,12 +137,30 @@ build-cni-plugin-image *args='--load':
         --file=Dockerfile-cni-plugin \
         --tag={{ cni-plugin-image }} \
         {{ args }}
-##
-## Test cluster
-##
 
+# Build docker image for cni-plugin-tester (Development)
+build-cni-plugin-test-image *args='--load':
+    docker buildx build . \
+        --file=cni-plugin/integration/flannel/Dockerfile-tester \
+        --tag={{ _cni-plugin-test-image }} \
+        {{ args }}
+
+# Build and load images for cni-plugin
+cni-plugin-test-integration-deps: build-cni-plugin-image build-cni-plugin-test-image _k3d-ready
+    @just-k3d import {{ _cni-plugin-test-image }} {{ cni-plugin-image }}
+
+# Run cni-plugin integration tests after preparing dependencies
+cni-plugin-test-integration: cni-plugin-test-integration-deps cni-plugin-test-integration-run
+
+# Run integration tests without preparing dependencies
+cni-plugin-test-integration-run:
+    TEST_CTX="k3d-$(just-k3d --evaluate K3D_CLUSTER_NAME)" ./cni-plugin/integration/run.sh
+
+
+
+# TODO(stevej): add a k3d-create-debug
 export K3S_DISABLE := "local-storage,traefik,servicelb,metrics-server@server:*"
-export K3D_CREATE_FLAGS := '--no-lb'
+export K3D_CREATE_FLAGS := '--no-lb --k3s-arg "--debug@server:*"'
```

Comment:
I need to move this into a flag for creating a cluster with debug logging.

Reply:
Hmmm, I have in mind perhaps two options. Both of them involve creating a cluster with `--debug` by default. I think it makes sense to have it on my default and then override it in CI (if we want to). Maybe this particular way of doing things wouldn't be a good fit for the other linkerd repos but in this case I think it makes sense, especially since we are going to be pretty reliant on kubelet logs for debugging. Anyway:

1) We could use something like: 

```
_default_k3d_flags := '--no-lb --k3s-arg "--debug@server:*"'
export K3D_CREATE_FLAGS := env_var_or_default("K3D_CREATE_FLAGS", _default_k3d_flags)
```

One of way of doing this is to export the environment variable based on a recipe-local temp var. Sounds like a mouthful but it means that by default we will use `--debug`. We can then override this either locally:

```
$ just _default_k3d_flags='--no-lb' k3d-create
```

or in CI using the same syntax.

Another approach here is to simply hardcode the default value in `K3D_CREATE_FLAGS`, locally and CI we can always override by setting the env variable.

2) We could have a recipe that handles this for us, e.g `_k3d_create_init` or something similar. We'd make use of a variable again to store default config and then export K3D_CREATE_FLAGS to the rest of the environment. This is more in line with what you're suggesting, but I think it just adds a bit more indirection when we can solve it maybe in a simpler way if we use `env_var_or_default`.

I'm by no means a just or makefile expert tho so lmk what you think.

Reply:
I turned this into an issue in the kanban board, number 65 there.